internal/exec/stages/files/files.go

```diff
@@ -145,20 +145,15 @@ type dirEntry types.Directory
 func (tmp dirEntry) create(l *log.Logger, u util.Util) error {
 	d := types.Directory(tmp)
 
-	if d.User.ID == nil {
-		d.User.ID = internalUtil.IntToPtr(0)
-	}
-	if d.Group.ID == nil {
-		d.Group.ID = internalUtil.IntToPtr(0)
-	}
+	d.User.ID, d.Group.ID = u.GetUserGroupID(l, d.User, d.Group)
 
 	err := l.LogOp(func() error {
 		path := filepath.Clean(u.JoinPath(string(d.Path)))
 
 		// Build a list of paths to create. Since os.MkdirAll only sets the mode for new directories and not the
 		// ownership, we need to determine which directories will be created so we don't chown something that already
 		// exists.
-		newPaths := []string{}
+		newPaths := []string{path}
 		for p := path; p != "/"; p = filepath.Dir(p) {
```

Comment:
This line should be `for p := filepath.Dir(path); ...`, otherwise if `path` doesn't exist it'll end up in `newPaths` twice.